server/modules/category/service/categoryImportJob.js

```diff
@@ -0,0 +1,202 @@
+    }
+  }
+
+  async _importLevels () {
+    this.logDebug('importing levels')
+
+    const categoryUuid = CategoryImportJobParams.getCategoryUuid(this.params)
+    const summary = CategoryImportJobParams.getSummary(this.params)
+
+    const surveyId = this.getSurveyId()
+    const user = this.getUser()
+
+    let category = await CategoryManager.deleteLevelsByCategory(user, surveyId, categoryUuid, this.tx)
+
+    const levelNames = CategoryImportSummary.getLevelNames(summary)
+
+    for (const levelName of levelNames) {
+      const levelToInsert = Category.newLevel(category, {
+        [CategoryLevel.props.name]: levelName
+      })
+      const level = await CategoryManager.insertLevel(user, surveyId, levelToInsert, this.tx)
+      category = Category.assocLevel(level)(category)
+    }
+
+    this.logDebug(`levels importing: ${levelNames}`)
+
+    return category
+  }
+
+  async _importItems (category) {
+    this.logDebug('importing items')
+
+    const summary = CategoryImportJobParams.getSummary(this.params)
+
+    this.logDebug('summary', summary)
+
+    const levels = Category.getLevelsArray(category)
+
+    const reader = await CategoryImportCSVParser.createRowsReader(
+      summary,
+      async itemRow => {
+        if (this.isCanceled()) {
+          reader.cancel()
+          return
+        }
+        await this._importRow(itemRow, levels)
+      },
+      total => this.total = total
+    )
+
+    await reader.start()
+  }
+
+  async _importRow (itemRow, levels) {
+    const { levelIndexDeeper, codes, labels, descriptions, extra } = itemRow
+
+    if (this._checkCodesNotEmpty(codes)) {
+      for (let levelIndex = 0; levelIndex <= levelIndexDeeper; levelIndex++) {
+        const level = levels[levelIndex]
+        const levelName = CategoryLevel.getName(level)
+
+        const codesLevel = codes.slice(0, levelIndex + 1)
+
+        const alreadyInserted = !!this.itemsByCodes[codesLevel]
+
+        if (alreadyInserted) {
+          if (levelIndex === levelIndexDeeper) {
+            this._addErrorCodeDuplicate(codesLevel)
+          }
+        } else {
+          let itemParentUuid = null
+          if (levelIndex > 0) {
+            const codesParent = codesLevel.slice(0, levelIndex)
+
+            const itemParent = this.itemsByCodes[codesParent]
+            itemParentUuid = CategoryItem.getUuid(itemParent)
+          }
+
+          // this.logDebug('inserting item', codesLevel)
+
+          const codeLevel = codesLevel[codesLevel.length - 1]
+
+          this.itemsByCodes[codesLevel] = CategoryItem.newItem(
+            CategoryLevel.getUuid(level),
+            itemParentUuid,
+            {
+              [CategoryItem.props.code]: codeLevel,
+              [ObjectUtils.keysProps.labels]: labels[levelName],
+              [ObjectUtils.keysProps.descriptions]: descriptions[levelName],
+              [CategoryItem.props.extra]: extra,
+            }
+          )
+        }
+      }
+    }
+    this.incrementProcessedItems()
```

Comment:
extra are still not imported. will you fix this in the next branch or this one?

Reply:
in the next branch